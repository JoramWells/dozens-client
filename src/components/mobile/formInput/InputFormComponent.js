import React, { useState } from "react";
import PropTypes from "prop-types";
import { validateInput } from "../../utilities/Validator";

const InputFormComponent = ({
  placeholder,
  Icon,
  validators,
  type,
  value,
  onChange,
}) => {
  const [error, setError] = useState(false);
  const handleChange = (event) => {
    const { value } = event.target;
    setError(validateInput(validators, value));
    onChange(value);
  };
  return (
    <div data-testid="input_form">
      <div
      style={{width:"18rem"}}
        className="bg-white w-full my-4 flex flex-row  ring-1 ring-gray-100 p-1  rounded-md " 
      >
        {Icon && <Icon className="h-5 m-1 text-gray-500" />}

        <input
          type={type}
          placeholder={placeholder}
          value={value}
          className="w-full focus:outline-none text-gray-700 rounded-md p-1"
          onChange={handleChange}
        />
      </div>
      <div className="h-2 flex justify-start">
        {error && <p className="text-xs text-red-400 -mt-1">{error.message}</p>}
      </div>
    </div>
  );
};
InputFormComponent.propTypes = {
  value: PropTypes.string,
  placeholder: PropTypes.string,
  type: PropTypes.string,
  validators: PropTypes.array,
  onChange: PropTypes.func,
  Icon: PropTypes.func,
};
InputFormComponent.defautlProps = {
  value: "",
  placeholder: "",
  type: "",
  validators: [],
  onChange: "",
  Icon: "",
};
export default InputFormComponent;
