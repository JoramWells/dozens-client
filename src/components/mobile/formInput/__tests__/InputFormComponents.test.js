import ReactDOM from "react-dom";
import InputFormComponent from "../InputFormComponent";
import { render, cleanup, screen } from "@testing-library/react";
import { MailIcon } from "@heroicons/react/solid";
import renderer from "react-test-renderer";
import { Validators } from "../../../utilities/Validator";

afterEach(() => {
  cleanup();
});
test("inputform component renders without crashing", () => {
  const form = document.createElement("form");
  ReactDOM.render(<InputFormComponent />, form);
});

test("inputformcomponent is in the dom", () => {
  const { getByTestId } = render(
    <form>
      <InputFormComponent />
    </form>
  );
  expect(getByTestId("input_form")).toBeInTheDocument();
});

test("inputformcomponetn should handle props correctly", () => {
  const handleChange = () => {
    console.log("Changed");
  };
  const props = {
    placeholder: "john doe",
    Icon: MailIcon,
    type: "text",
    value: "Joram Bramuel",
    onChange: handleChange,
  };

  render(
    <InputFormComponent
      placeholder={props.placeholder}
      Icon={props.Icon}
      type={props.type}
      value={props.value}
      onChange={props.onChange}
      validators={[
        {check:Validators.email, message:"Invalid email"}
      ]}
    />
  );
  const formElement = screen.getByTestId("input_form");
  expect(formElement).toBeInTheDocument();
});

test("inputformcomponent matches snapshot", () => {
  const handleChange = () => {
    console.log("Changed");
  };
  const props = {
    placeholder: "john doe",
    Icon: MailIcon,
    type: "text",
    value: "Joram Bramuel",
    onChange: handleChange,
  };
  const tree = renderer
    .create(
      <InputFormComponent
        placeholder={props.placeholder}
        Icon={props.Icon}
        type={props.type}
        value={props.value}
        onChange={props.onChange}
        validators={[
          {check:Validators.email, message:"Invalid email"}
        ]}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
//   tree.props._onChange();
});
