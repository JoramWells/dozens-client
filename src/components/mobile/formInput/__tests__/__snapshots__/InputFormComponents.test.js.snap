// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inputformcomponent matches snapshot 1`] = `
<div
  data-testid="input_form"
>
  <div
    className="bg-white w-full my-4 flex flex-row  ring-1 ring-gray-100 p-1  rounded-md "
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <svg
      className="h-5 m-1 text-gray-500"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2.003 5.884L10 9.882l7.997-3.998A2 2 0 0016 4H4a2 2 0 00-1.997 1.884z"
      />
      <path
        d="M18 8.118l-8 4-8-4V14a2 2 0 002 2h12a2 2 0 002-2V8.118z"
      />
    </svg>
    <input
      className="w-full focus:outline-none text-gray-700 rounded-md p-1"
      onChange={[Function]}
      placeholder="john doe"
      type="text"
      value="Joram Bramuel"
    />
  </div>
  <div
    className="h-2 flex justify-start"
  />
</div>
`;
